### PR TITLE
Robots new app config input types

### PR DESCRIPTION
**Pull request by poi33**

### CODE REVIEW

File: src/main/resources/services/robotstxt/robotstxt.js [hunk 1]
```diff
@@ -1,13 +1,122 @@
...
+
+    if (Array.isArray(group.allow)) {
+        group.allow.forEach(function (allow) {
+            body += `Allow: ${allow}\n`;
+        });
+    } else {
+        body += `Allow: ${group.allow}\n`;
+    }
+
+    if (Array.isArray(group.disallow)) {
```
Comment (reisfmb) [inline src/main/resources/services/robotstxt/robotstxt.js:65]: Suggestion... create a function:

```javascript
function composeBody(arr) {
    const forceArray = (el) => Array.isArray(el) ? el : [el];
    const keyValuesToString = (obj) => forceArray(obj.values).map(value => `${obj.key}: ${value}\n`).join('');
    
    return arr.reduce((acc, curr) => acc + keyValuesToString(curr), '');
}
```

and use it like:

```javascript
const body = composeBody([
    { key: "User-agent", values: group.userAgent },
    { key: "Allow", values: group.allow },
    { key: "Disallow", values: group.disallow }
]);
```

**Review by reisfmb — APPROVED** [review]

File: src/main/resources/services/robotstxt/robotstxt.js [hunk 2]
```diff
@@ -1,13 +1,124 @@
...
 function handleGet(req) {
 
-    var siteConfig = libs.portal.getSiteConfig();
-    return {
-        contentType: 'text/plain',
-        body: siteConfig.robotstxt
-    };
+    const siteConfig = libs.portal.getSiteConfig();
+
+    const accept = req.headers.accept || req.headers.Accept;
```
Comment (alansemenov) [inline src/main/resources/services/robotstxt/robotstxt.js:9]: `Request` object has `getHeader` method which can be used to get the Accept header. Header name parameter is case-insensitive (can be `req.getHeader('accept')` or `req.getHeader('Accept')`).

https://github.com/enonic/site-developer/blob/master/src/main/resources/site/parts/market-latest/market-latest.es#L89

Comment (alansemenov) [reply src/main/resources/services/robotstxt/robotstxt.js:9]: Documented [here](https://developer.enonic.com/docs/xp/stable/framework/http#http-request) (towards the bottom)

File: src/main/resources/services/robotstxt/robotstxt.js [hunk 3]
```diff
@@ -1,13 +1,124 @@
...
+        return data;
+    } else {
+        const data = {
+            rules: getGroupData(config.groups)
+        }
+        if (config.sitemap) {
+            data.sitemap = config.sitemap;
+        }
+
+        return data;
```
Comment (alansemenov) [inline src/main/resources/services/robotstxt/robotstxt.js:114]: Enough with one `return data;` after the `if...` block.

**Review by alansemenov — CHANGES_REQUESTED** [review]
A couple of cosmetic things to fix

File: src/main/resources/services/robotstxt/robotstxt.js (hunk 1 above)
Comment (poi33) [reply src/main/resources/services/robotstxt/robotstxt.js:65]: Let me see if I can adopt this with a some modification

**Review by alansemenov — APPROVED** [review]

**Pull request merged**